tests/test-admin-deploy-karg.sh

```diff
@@ -44,6 +44,7 @@ echo "ok deploy with --karg, but same config"
 
 ${CMD_PREFIX} ostree admin deploy --karg-proc-cmdline --os=testos testos:testos/buildmaster/x86_64-runtime
 for arg in $(cat /proc/cmdline); do
+    if test "${arg#ostree=*}" != ${arg}; then continue; fi
```

Comment:
I would probably write this as a `case` statement. Something like
```
case "$arg" in
  ostree=*)
    # Skip ostree arg that gets stripped out
    ;;
  *)
    assert_file_has_content ...
    ;;
esac
```
But this seems fine, too. I'll try to give it a local test.